src/Bicep.LangServer.UnitTests/Handlers/BicepCodeActionHandlerTests.cs

```diff
@@ -62,7 +63,7 @@ public async Task VerifyEditRuleInBicepCodeActionIsAvailable_WhenClientSupportsS
             clientCapabilitiesProvider.Setup(m => m.DoesClientSupportShowDocumentRequest()).Returns(clientSupportShowDocumentRequestAndWorkspaceFolders);
             clientCapabilitiesProvider.Setup(m => m.DoesClientSupportWorkspaceFolders()).Returns(clientSupportShowDocumentRequestAndWorkspaceFolders);
 
-            var bicepEditLinterRuleHandler = new BicepCodeActionHandler(bicepCompilationManager, clientCapabilitiesProvider.Object);
+            var bicepEditLinterRuleHandler = new BicepCodeActionHandler(bicepCompilationManager, clientCapabilitiesProvider.Object, new DocumentSelectorFactory(null));
```

Comment:
Rather than making this nullable, what about having a `.Default` property on `BicepLangServerOptions`? If you use a `record`, you get the benefit of it being immutable and can use `with` for convenience.

e.g.
```csharp
public record BicepLangServerOptions(
    bool VsCompatibilityMode)
{
    public static BicepLangServerOptions Default { get; } = new(
        VsCompatibilityMode: false);
}
```

Could then be used as:
```csharp
var options = BicepLangServerOptions.Default with { VsCompatibilityMode = true };
```

Reply:
7a833789b80fae27020f7397d59a88f7cbf68618